model-optimizer/extensions/front/tf/CTCLossReplacement.py

```diff
@@ -0,0 +1,183 @@
+                ('seq_len', 'ctc_loss', {'out': 0, 'in': 3}),
+                ('transpose', 'ctc_greedy_decoder', {'out': 0, 'in': 0}),
+                ('transpose', 'ctc_loss', {'out': 0, 'in': 0}),
+                ('ctc_greedy_decoder', 'sparse_to_dense', {'out': 0, 'in': 0}),
+                ('ctc_greedy_decoder', 'sparse_to_dense', {'out': 2, 'in': 1}),
+                ('ctc_greedy_decoder', 'sparse_to_dense', {'out': 1, 'in': 2}),
+                ('const', 'sparse_to_dense', {'out': 0, 'in': 3}),
+                ('ctc_greedy_decoder', 'cast', {'out': 1, 'in': 0}),
+                ('ctc_greedy_decoder', 'ctc_loss', {'out': 0, 'in': 1}),
+                ('cast', 'ctc_loss', {'out': 0, 'in': 2})
+            ])
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        seq_len_tf = match['seq_len']
+        transpose_tf = match['transpose']
+        ctc_greedy_decoder_tf = match['ctc_greedy_decoder']
+        cast_tf = match['cast']
+        ctc_loss_tf = match['ctc_loss']
+        sparse_to_dense_tf = match['sparse_to_dense']
+
+        output_sparse_to_dense_name = sparse_to_dense_tf.soft_get('name', sparse_to_dense_tf.id)
+        output_ctc_loss_name = ctc_loss_tf.soft_get('name', ctc_loss_tf.id)
+        ctc_greedy_decoder_tf_name = ctc_greedy_decoder_tf.soft_get('name', ctc_greedy_decoder_tf.id)
+
+        log.debug('Found CTCLossFrontReplacer pattern after {} with name {}'.format(ctc_greedy_decoder_tf.op,
+                                                                                    ctc_greedy_decoder_tf.name))
+
+        # create sequence mask node, sub-graph for transforming into sequence length and connect with consumers
+        seq_len_tf_shape = seq_len_tf.soft_get('shape', None)
+        if seq_len_tf_shape is None or len(seq_len_tf_shape) != 2:
+            raise Error('The sequence length that is the second input to the CTCGreedyDecoder node "{}"'
+                        ' must be specified in a mask format.'.format(ctc_greedy_decoder_tf_name))
+        seq_len_tf_type = seq_len_tf.soft_get('data_type', None)
+        seq_len_tf_name = seq_len_tf.soft_get('name', seq_len_tf.id)
+        seq_mask_placeholder = Parameter(graph, {'name': seq_len_tf_name, 'shape': seq_len_tf_shape,
+                                                 'data_type': seq_len_tf_type}).create_node()
+        reduce_to_seq_len_node = create_op_with_const_inputs(graph, ReduceSum, {1: np.array(1, dtype=np.int32)},
+                                                             {'name': seq_len_tf_name + '/ReduceToSeqLen',
+                                                              'keep_dims': False})
+        reduce_to_seq_len_node.in_port(0).connect(seq_mask_placeholder.out_port(0))
+        seq_len_tf.out_port(0).get_connection().set_source(reduce_to_seq_len_node.out_port(0))
+
+        cast_fp_type = np.float32
+        if 'data_type' in graph.graph['cmd_params']:
```

Comment:
nit: The `data_type` is always in the `cmd_params`, so you can safely use the value directly.

Reply:
Fixed.